src/EditorFeatures/Core/Tagging/AbstractAsynchronousTaggerProvider.TagSource_ProduceTags.cs

```diff
@@ -374,31 +374,20 @@ private void CheckSnapshot(ITextSnapshot snapshot)
 
             private ImmutableDictionary<ITextBuffer, TagSpanIntervalTree<TTag>> ConvertToTagTrees(
                 ImmutableDictionary<ITextBuffer, TagSpanIntervalTree<TTag>> oldTagTrees,
+                ISet<ITextBuffer> buffersToTag,
                 ILookup<ITextBuffer, ITagSpan<TTag>> newTagsByBuffer,
                 IEnumerable<DocumentSnapshotSpan> spansTagged)
             {
-                // NOTE: we assume that the following list is already realized and is _not_ lazily
-                // computed. It's not clear what the contract is of this API.
-
-                // common case where we only tagged a single range of a document.
-                if (spansTagged.IsSingle())
-                {
-                    return ConvertToTagTree(oldTagTrees, newTagsByBuffer, spansTagged.Single().SnapshotSpan);
```

Comment:
[Questions which can be followed up later:]

I believe ConvertToTagTree is now dead? Should ComputeNewTagTree still call the special case overload if spansToInvalidateByBuffer gives buffer with a single span?

Reply:
I could go either way. For now the code is updated to eliminate all the code which is now dead.